### PR TITLE
Add batched solve_triangular

### PR DESCRIPTION
**Pull request by Berrysoft**
It is not implemented in scipy, but I need it for my research:)

### CODE REVIEW

File: cupyx/scipy/linalg/_solve_triangular.py [hunk 1]
```diff
@@ -35,27 +35,37 @@ def solve_triangular(a, b, trans=0, lower=False, unit_diagonal=False,
 
     Returns:
         cupy.ndarray:
-            The matrix with dimension ``(M,)`` or ``(M, N)``.
+            The matrix with dimension ``(..., M,)`` or ``(..., M, N)``.
```
Comment (kmaehashi) [inline cupyx/scipy/linalg/_solve_triangular.py:39]:
```suggestion

    .. note::
       Unlike the SciPy counterpart, the CuPy implementation supports batches
       of matrices.
```

**Review by kmaehashi — COMMENTED** [review]
Hi @Berrysoft, thanks for the contribution! We usually keep our APIs compatible with SciPy ones but this sounds good addition without any breakage.

**Comment by Berrysoft** [comment]
The hip used by CI seems different from mine. I'll fix it later.

**Comment by takagi** [comment]
record: https://ci.preferred.jp/p/dashboard_by_commit_id?commit_id=498ce858480c19b5c7ebf46207438aae28db4a8b

**Comment by takagi** [comment]
The failure seems not to be related.

**Review by takagi — APPROVED** [review]
LGTM!

**Pull request merged**